exploration/iri_seasons.md

```diff
@@ -0,0 +1,208 @@
+da.plot(ax=ax)
+```
+
+## Process data
+
+Not the most computationally efficient, but works for this size of data.
+
+We just interpolate the IRI forecast at the coordinates of the "inseason".
+Then we can re-aggregate however we want (adm0, adm1, livelihoods zone, etc).
+
+```python
+# utils.process_iri_inseason()
+```
+
+```python
+# check plots
+
+da = utils.load_iri_inseason(forecast_date="2020-02-16")
+
+for L in range(1, 5):
+    fig, ax = plt.subplots(figsize=(25, 5))
+    da.sel(L=L).plot(ax=ax)
+```
+
+```python
+# utils.calculate_iri_inseason_stats()
+```
+
+```python
+stats = utils.load_iri_inseason_stats()
+stats["rel_month1"] = stats["rel_month1"].apply(
+    lambda x: x if x < 13 else x - 12
+)
+
+# calculate fraction of pixels in AOI that might have a season
+# i.e., ones that don't have an error code ( > 250) in ASAP rasters
+
+# note: must do as float, otherwise compute_raster_stats doesn't work
+s_pos_da = xr.where(utils.load_asap_inseason("dekad", 1) < 251, 1.0, 0.0)
+s_pos = s_pos_da.oap.compute_raster_stats(aoi, feature_col="ADM0_CODE")
+s_pos = s_pos.rename(
+    columns={"sum": "possible_season_pixels", "count": "total_pixels"}
+)
+
+# fraction with possible season is close to 100% for BFA, but less for TCD and NER
+display(s_pos)
+
+stats = stats.merge(
+    s_pos[["ADM0_CODE", "possible_season_pixels", "total_pixels"]],
+    on="ADM0_CODE",
+)
+stats["frac_reporting"] = stats["count"] / stats["possible_season_pixels"]
+display(stats)
+```
+
+## Selecting relevant dates
+
+We can select relevant dates either by:
+
+- Just picking which months we care about
+  - Based on previous frameworks this would be Jun - Oct
+- Only monitoring once the fraction of "reporting" pixels
+goes above a certain level, say 50%
+
+```python
+# check typical values by leadtime, month
+
+agg_cols = ["L", "rel_month1"]
+plot_cols = ["30quant", "frac_reporting"]
+
+for agg_col in agg_cols:
+    year_group = stats.groupby(["F_year", agg_col])[plot_cols].mean()
+    dff = year_group.groupby(agg_col)[plot_cols].agg(["max", "mean", "min"])
+    for plot_col in plot_cols:
+        fig, ax = plt.subplots()
+        dff[plot_col].plot(ax=ax)
+        ax.set_ylabel(plot_col)
+
+# note that rel_month1 is the first relevant month of the forecast
+
+# things are pretty stable by leadtime, except fewer tend to be reporting
+# at L=4
+
+# we see that if we do not select which months we care about,
+# we will often end up triggering in October
+```
+
+Unfortunately `frac_reporting` is still pretty high in September.
+This means that we could trigger for a forecast relevant for Sep-Oct-Noc,
+which doesn't seem right.
+
+Basically this means that we need to either:
+
+- Threshold based on how many dekads within the trimester each pixel is in season
+  - We are currently just using `any()` (i.e. the trimester is inseason
+if any of its dekads are inseason)
+- Use ECMWF which forecasts monthly instead of trimesterially
```

Comment:
I would recommend this, not sure why we don't use ECMWF anyway given its historical record. I was saying this to Zach the other day, I don't really get why we need tercile probabilities if they are being converted into a classification anyway, and at least with ECMWF we have a historical record for performance validation. Thoughts? I think I'll put a message in the Slack.

Reply:
Very much agreed! I'll drop a message in Slack too, but Zack and I are both working on using ECMWF for Dry Corridor and Sahel respectively.